tsdb/shard.go

```diff
@@ -760,11 +761,7 @@ func (s *Shard) createFieldsAndMeasurements(fieldsToCreate []*FieldCreate) error
 		s.index.SetFieldName(f.Measurement, f.Field.Name)
 	}
 
-	if len(fieldsToCreate) > 0 {
```

Comment:
This seems like a good optimisation - ditching it seems bad.

Reply:
It is checked as the first statement in the procedure. This second check looks redundant to me.